migration-engine/cli/tests/cli_tests.rs

```diff
@@ -311,6 +311,53 @@ fn basic_jsonrpc_roundtrip_works(_api: TestApi) {
 }
 
 #[test]
+fn introspect_empty_database() {
+    use std::io::{BufRead, BufReader, Write as _};
+    let tmpdir = tempfile::tempdir().unwrap();
+    let schema = r#"
+        datasource db {
+            provider = "sqlite"
+            url = env("TEST_DATABASE_URL")
+        }
+
+    "#;
+    std::fs::File::create(tmpdir.path().join("dev.db")).unwrap();
+    let mut process = Command::new(migration_engine_bin_path())
+        .env(
+            "TEST_DATABASE_URL",
+            format!("file:{}/dev.db", tmpdir.path().to_string_lossy()),
+        )
+        .stdin(std::process::Stdio::piped())
+        .stdout(std::process::Stdio::piped())
+        .stderr(std::process::Stdio::null())
+        .spawn()
+        .unwrap();
+    let stdin = process.stdin.as_mut().unwrap();
+    let mut stdout = BufReader::new(process.stdout.as_mut().unwrap());
+
+    let msg = serde_json::to_string(&serde_json::json!({
+        "jsonrpc": "2.0",
+        "method": "introspect",
+        "id": 1,
+        "params": {
+            "schema": schema,
+            "force": true,
+            "compositeTypeDepth": 5,
+        }
+    }))
+    .unwrap();
+    stdin.write_all(msg.as_bytes()).unwrap();
+    stdin.write_all(b"\n").unwrap();
+
+    let mut response = String::new();
+    stdout.read_line(&mut response).unwrap();
+
+    assert!(response.starts_with(r##"{"jsonrpc":"2.0","error":{"code":4466,"message":"An error happened. Check the data field for details.","data":{"is_panic":false,"message":"The introspected database was empty.","meta":null,"error_code":"P4001"}},"id":1}"##));
+}
+
+// TODO: create a basic table before introspecting
+#[test]
+#[ignore]
```

Comment:
We try hard to avoid `#[ignore]` on tests, I'd rather we deleted it if we really can't do it.